pkg/action/custom_resource_lister.go

```diff
@@ -0,0 +1,138 @@
+package action
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/api/pkg/operators/v2alpha1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// FindOperator finds an operator object on-cluster provided a package and namespace.
+func FindOperator(ctx context.Context, client client.Client, key types.NamespacedName) (*v2alpha1.Operator, error) {
+	operator := v2alpha1.Operator{}
+
+	err := client.Get(ctx, key, &operator)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, fmt.Errorf("package %s not found", key.Name)
+		}
+		return nil, err
+	}
+	return &operator, nil
+}
+
+// Unzip finds the CSV referenced by the provided operator and then inspects the spec.customresourcedefinitions.owned
+// section of the CSV to return a list of APIs that are owned by the CSV.
+func Unzip(ctx context.Context, client client.Client, operator *v2alpha1.Operator) ([]v1alpha1.CRDDescription, error) {
```

Comment:
For now, I'd suggest making `FindOperator`, `Unzip`, `List`, and `ListAll` unexported functions on the `OperatorListOperands` struct.